app/commands/exercise/search.rb

```diff
@@ -10,11 +10,21 @@ def initialize(track, user_track: nil, criteria: nil)
 
     def call
       @exercises = track.exercises
+      filter_status!
       filter_criteria!
       sort!
       @exercises
     end
 
+    def filter_status!
+      if !user_track || user_track.external?
+        @exercises = @exercises.where.not(status: %i[deprecated wip])
+      else
+        @exercises = @exercises.where.not(status: %i[deprecated wip]).
+          or(@exercises.where(id: user_track.solutions.pluck(:exercise_id)))
+      end
```

Comment:
I had to duplicate the `where.not` bit for Rails to correctly pick up the `.or`, but maybe there is a better way?